### ISSUE TITLE
Model definition not found in default responses

### ISSUE THREAD

**Issue by hassansin** [swagger-api/swagger-ui#1700]
I've this `test.json` schema where a  `ResultObject` response object references a `Result` model in definitions:

```
{
  "swagger": "2.0",  
  "host": "localhost:9000",
  "schemes": [
    "http"
  ],
  "responses": {
    "ResultObject": {
      "description": "Result object.",
      "schema": {
        "$ref": "#/definitions/Result"
      }
    }
  },
  "paths": {
    "/sheet": {
      "delete": {
        "summary": "Delete Column.",
        "tags": [
          "Column"
        ],
        "responses": {
          "200": {
            "$ref": "#/responses/ResultObject"
          }
        },
        "parameters": [ ]
      }      
    }    
  },
  "definitions": {    
    "Result": {
      "type": "object",
      "properties": {
        "resultCode": {
          "type": "number"         
        }        
      }
    }
  }
}
```

In UI, I get the error `http://localhost:9000/spec/test.json#/definitions/Result is not defined!`: 

![image](https://cloud.githubusercontent.com/assets/4618044/10715762/3236307c-7b4c-11e5-9169-24e79a8477c9.png)

Is it swagger-ui bug or my schema is invalid?

**Comment by hassansin** [comment]
It seems to work on swagger-editor:

![image](https://cloud.githubusercontent.com/assets/4618044/10715777/9cff0c30-7b4c-11e5-8282-70a3c7a0eea4.png)

**Comment by texnofobix** [comment]
I also can confirm this behavior. I have seen it happen in both yaml and json formats and with absolute and relative pathing.

The error reported in Chrome shows:
```
 Uncaught TypeError: Cannot read property 'definitions' of nullResolver.resolve @ schema-markup.js:302(anonymous function) @ resolver.js:268SwaggerSpecConverter.finish @ operation.js:838SwaggerSpecConverter.convert @ operation.js:237obj.on.response @ resolver.js:266responseInterceptor @ schema-markup.js:50obj.on.response @ schema-markup.js:54(anonymous function) @ schema-markup.js:248Request.callback @ index.js:24(anonymous function) @ index.js:24Emitter.emit @ index.js:24xhr.onreadystatechange @ index.js:24
```

**Comment by fehguy** [comment]
Hi, this is a swagger-js issue--the shared `responses` object is not implemented.  Can you please open a ticket in that project and we can get it resolved?

**Comment by hassansin** [comment]
Opened swagger-api/swagger-js#606

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen if you experience otherwise.
